src/prefect/new_task_engine.py

```diff
@@ -376,10 +382,10 @@ async def run_task_async(
 
     # This is a context manager that keeps track of the run of the task run.
     with engine.start() as run:
-        run.begin_run()
+        with run.enter_run_context():
+            run.begin_run()
 
-        while run.is_running():
-            with run.enter_run_context():
+            while run.is_running():
```

Comment:
this may not be what we want, but this is to address the fact that `task_input_hash` uses the `TaskRunContext`, which would not exist unless we first enter the context

Reply:
I think some of @cicdw's work may preclude this, pinging in case